src/main/cpp/util/profiler_posix.cc

```diff
@@ -0,0 +1,33 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/main/cpp/util/profiler.h"
+
+#include <time.h>
+
+namespace blaze_util {
+namespace profiler {
+
+Ticks Ticks::Now() {
+  struct timespec ts = {};
+  clock_gettime(CLOCK_MONOTONIC, &ts);
+  return {ts.tv_sec * 1000000000LL + ts.tv_nsec};
```

Comment:
Why do you count nanoseconds if you are only interested in milliseconds at the end? AFAIU 2^63 nanoseconds is still a lot of time (centuries), but still.

Reply:
Because then we don't have to divide `tv_nsec`, and don't lose precision. Not that either of those is significant though.

Reply:
However, since the resolution is 1ns, and 2^63 still gives enough range, I'm keeping it at nanoseconds.